### PR TITLE
added convenience func that increments a tag by exactly 1

### PR DESCRIPTION
**Pull request by caquillo07**

### CODE REVIEW

File: statsd.go [hunk 1]
```diff
@@ -83,6 +83,11 @@ func Incr(name string, tags []string, rate float64) {
 	}
 }
 
+// IncrByOne convenience function that increments the tags by exactly 1
+func IncrByOne(name string, tags ...string) {
+	IncrByOne(name, tags...)
```
Comment (acj) [inline statsd.go:88]: Did you mean to call `Incr`?

**Review by acj — COMMENTED** [review]
I think `Incr` already does what you want? The `rate` parameter is the sampling rate ([docs](https://docs.datadoghq.com/metrics/custom_metrics/dogstatsd_metrics_submission/?code-lang=go#functions)), not the amount to increment. (Kinda confusing, tbh, and could be renamed.)

File: statsd.go (hunk 1 above)
Comment (caquillo07) [reply statsd.go]: 🤦🏼

File: statsd.go [hunk 2]
```diff
@@ -12,9 +12,10 @@ var client *statsd.Client
 var log Logger
 
 func init() {
-	Disable()
+	_ = Disable()
```
Comment (caquillo07) [inline statsd.go:15]: are we okay dropping this error here?

**Review by nharsh — COMMENTED** [review]
I could go either way on adding this function. I don't think it saves us a lot of typing but there's not much overhead in adding it so if people find it useful then we should do it.

I do think that the comments that are added here for every function aren't necessary. Most of these functions are short and their intention seems clear already.

**Comment by acj** [comment]
Does my initial comment seem right? I think the new function is equivalent to `Incr`, just without the sampling rate argument. The original function increments by exactly 1.

Upstream func: https://github.com/DataDog/datadog-go/blob/553de96e699a42be8b401607fbbbce81d4942790/statsd/statsd.go#L174-L175

> I do think that the comments that are added here for every function aren't necessary. Most of these functions are short and their intention seems clear already.

+1. The linter probably wants comments because these functions are exported, but I don't think there's a strong need.

**Comment by caquillo07** [comment]
> I could go either way on adding this function. I don't think it saves us a lot of typing but there's not much overhead in adding it so if people find it useful then we should do it.
> 
> I do think that the comments that are added here for every function aren't necessary. Most of these functions are short and their intention seems clear already.

Yea the comments here are purely from the formatter :) in Go is expected that every exported type/function comes with a comment explaining what it does, its mostly required for building godocs; its also nice to peek at the comments to see what it does

**Comment by caquillo07** [comment]
> Does my initial comment seem right? I think the new function is equivalent to `Incr`, just without the sampling rate argument. The original function increments by exactly 1.
> 
> Upstream func: https://github.com/DataDog/datadog-go/blob/553de96e699a42be8b401607fbbbce81d4942790/statsd/statsd.go#L174-L175
> 
> > I do think that the comments that are added here for every function aren't necessary. Most of these functions are short and their intention seems clear already.
> 
> +1. The linter probably wants comments because these functions are exported, but I don't think there's a strong need.

ahhh I did not go that far into the docs, but that makes a lot of sense. The idea behind this function is to be able to do the following:

```go
		statsd.IncrByOne("gen3_workflows", "status:failed", "workflow_name:thumbnail-from-video", "error_type:send_error_message_to_queue")

// vs 
		statsd.Incr("gen3_workflows", []string{"status:failed", "workflow_name:thumbnail-from-video", "error_type:send_error_message_to_queue"}, 1)
```

I noticed every single invocation of this function always has a `1`, and getting rid of that last argument let's use `...variadic` instead of a fixed string slice, which can be a bit tedious hah. Happy to remove if we don't see value in it

**Comment by caquillo07** [comment]
> Oh, gotcha. Yeah, that's a nice idea since we basically never change the sampling arg (afaik). There's prior art for making a simplified version of an existing function with different parameters (e.g. `SimpleMeasureExecTime`), so you could follow that if you want.
> 
> If we eventually want to change to the variadic versions, there will be a lot of places to update in different repos -- probably needs a bigger discussion.

Oh no, no need to refactor existing code haha this is just a convenience function for anything going forward - def not worth the looking back and changing :D

Also, love that name! will go with that `SimpleIncr`

**Review by acj — COMMENTED** [review]
A few more suggestions. LGTM otherwise.

File: statsd.go [hunk 3]
```diff
@@ -47,42 +49,54 @@ func Disable() error {
 	return nil
 }
 
+// Count tracks how many times something happened per second.
```
[the remark this answers is not in the document]
Comment (caquillo07) [reply statsd.go:52]: makes sense!

**Review by acj — APPROVED** [review]

**Pull request merged**